Add Bézier curve components (#603)
motion-canvas/motion-canvas: packages/docs/docs/components/bezier.mdx

import {makeScene2D} from '@motion-canvas/2d/lib/scenes';
import {QuadBezier} from '@motion-canvas/2d/lib/components';
import {createRef} from '@motion-canvas/core/lib/utils';
import {all} from '@motion-canvas/core/lib/flow';

export default makeScene2D(function* (view) {
  const bezier = createRef<Spline>();

  view.add(
    <QuadBezier
      ref={bezier}
      lineWidth={6}
      stroke={'lightseagreen'}
      p0={[-150, 50]}
      p1={[0, -120]}
      p2={[150, 50]}
      end={0}
    />,
  );

  yield* bezier().end(1, 1);
  yield* bezier().start(1, 1).to(0, 1);
});
```

Bézier curves are ubiquitous in computer graphics. Motion Canvas comes with
components to draw both quadratic as well as cubic Bézier curves.

:::tip

If you're trying to draw more complicated shapes than single Bézier curves allow
for, check out the [`Spline`][spline] component, instead.

:::

## Using the components

To draw Bézier curves, we can use the [`QuadBezier`][quad-bezier] component for
quadratic Bézier curves and [`CubicBezier`][cubic-bezier] component for cubic
Bézier curves, respectively.

:::info

Most of the examples on this page can be switched between cubic and quadratic

> Bézier curves by using the dropdown at the right side of the animation player.

Reviewer: ```suggestion
Bézier curves by using the dropdown on the right side of the animation player.
```

> To draw Bézier curves, we can use the [`QuadBezier`][quad-bezier] component for
> quadratic Bézier curves and [`CubicBezier`][cubic-bezier] component for cubic
> Bézier curves, respectively.
> 
> :::info
> 
> Most of the examples on this page can be switched between cubic and quadratic
> Bézier curves by using the dropdown at the right side of the animation player.
> 
> :::

Reviewer: I think this could be shortened to:
```suggestion
Each example below is applicable to both the [`QuadBezier`][quad-bezier] and
[`CubicBezier`][cubic-bezier] nodes. You can switch between the two types of
curves using the dropdown on the right side of the animation player.
```

### Defining control points

Bézier curves are defined by a start and end point, as well as several control
points. The exact number of control points is different for different kinds of

> Bézier curves. A quadratic Bézier curve only has a single control point whereas

Reviewer: ```suggestion
Bézier curves. A quadratic Bézier curve has only a single control point whereas
```

a cubic Bézier curve has two.

> To define a Bézier curve, we need to define its start and point, as well as its
> control points.

Reviewer: This is not necessary
```suggestion
```
You already said that in the previous paragraph:
```
Bézier curves are defined by a start and end point, as well as several control
points.
```
